src/projections.F90

```diff
@@ -544,32 +572,35 @@ subroutine point_volume(x0, tu, tv, tw, ku, kv, kw, coef, nctlu, nctlv, nctlw, n
                 ! If the constraints are applied - and the new point
                 ! doesn't satisfy the Wolfe conditions, it doesn't make
                 ! sense to apply a quadratic approximation
-                step = 0.25 * step
+                step = 0.1 * step
             else
-                ! c = nfval - fval - pgrad * step is always positive since
-                ! nfval - fval > pgrad * wolfe * step > pgrad * step
                 c = ((nfval - fval) - pgrad * step)
-                step = -step * step * pgrad / (2.0 * c)
+                quad_step = -step * step * pgrad / (2.0 * c)
                 ! This update is always less than the original step length
+                ! however, it might be too small... in that case,
+                ! just backtrack since the quad. approx is probably not too good.
+                if ((quad_step .lt. eps) .or. (quad_step .ne. quad_step)) then
+                    step = 0.1 * step
+                else
+                    step = quad_step
+                end if
             end if
+
         end do lineloop
 
-        if (m == nLine + 1) then
-            dist = ndist
-            nLSFail = nLSFail + 1
-            if (NLSFail > LSFailMax) then ! There is nothing more we can do...
-                exit Iteration_loop
-            end if
-        else
-            NLSFail = 0
-            ! Check if there has been no change in the coordinates
-            p_diff = NORM2(pt - newpt)
-            if (p_diff < eps) then
-                exit Iteration_loop
-            end if
-        end if
+        ! update the point
```

Comment:
It would cause difficult points to terminate early for no reason. The line search "failure" was based on it reaching the max iterations. @sseraj has several points that was quitting prematurely because of this check.